django/test/client.py

```diff
@@ -345,12 +345,16 @@ class RequestFactory:
     just as if that view had been hooked up using a URLconf.
     """
 
-    def __init__(self, *, json_encoder=DjangoJSONEncoder, **defaults):
+    def __init__(self, *, json_encoder=DjangoJSONEncoder, headers=None, **defaults):
         self.json_encoder = json_encoder
         self.defaults = defaults
         self.cookies = SimpleCookie()
         self.errors = BytesIO()
 
+        if headers:
+            transformed_headers = self._transform_headers(headers)
+            self.defaults.update(transformed_headers)
```

Comment:
I think we can avoid creating a temporary var here
```suggestion
            self.defaults.update(self._transform_headers(headers)
```

Reply:
Then, in combination with Adam's various comments this could become:
```python
            self.defaults.update(HttpHeaders.to_wsgi_names(headers))
```